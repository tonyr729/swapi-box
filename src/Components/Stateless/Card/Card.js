import React from 'react';
import './Card.css'

const Card =({data, setFavorites, favorites}) => {
  
  const keys = Object.keys(data)
  const cardContents = keys.map((keyname, index) => {
    let key = keyname;
    if(key === "vehicleClass") {
      key = "class"
    } 
    if (key === 'favorite'){
      return <p></p>
    }
    return (
      <p key={index} >{ key.charAt(0).toUpperCase() + key.slice(1) }: { data[keyname] }</p>
    )
  })

  const classValue = (data) => {
    let favs = favorites || [1, 2]
    let match = favs.find(favorite => favorite.name === data.name)

    return match ? "card-button active" : "card-button";
  }

  
    
  return (
    <div className="card">
      <button className={classValue(data)} onClick={() => setFavorites(data)}>â˜…</button>
      <div className='card-text'>
        { cardContents }
      </div>
    </div>
  )
}

export default Card;