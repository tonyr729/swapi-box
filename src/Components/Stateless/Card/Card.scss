.card {
  background-color: rgba(255, 255, 255, .3);
  color: rgb(255, 255, 255)
}