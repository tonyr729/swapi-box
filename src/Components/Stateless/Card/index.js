import React from 'react';
import './index.css';
import PropTypes from 'prop-types';

const Card =({data, setFavorites, favorites}) => {
  
  const keys = Object.keys(data);
  const cardContents = keys.map((keyname, index) => {
    let key = keyname;
    if(key === "vehicleClass") {
      key = "class";
    } 
    if (key === 'favorite'){
      return (<p key={index}></p>);
    }
    return (
      <p key={index} >{ key.charAt(0).toUpperCase() + key.slice(1) }: { data[keyname] }</p>
    );
  })

  const classValue = (data) => {
    let favs = favorites || [1, 2];
    let match = favs.find(favorite => favorite.name === data.name);

    return match ? "card-button active" : "card-button";
  }

  
    
  return (
    <div className="card">
      <button className={classValue(data)} onClick={() => setFavorites(data)}>â˜…</button>
      <div className='card-text'>
        { cardContents }
      </div>
    </div>
  )
}

Card.propTypes = {
  data: PropTypes.any,
  setFavorites: PropTypes.func,
  favorites: PropTypes.array
};

export default Card;