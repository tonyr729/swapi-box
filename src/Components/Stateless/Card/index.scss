.card {
  background-color: rgba(255, 255, 255, 0.3);
  color: white;
  margin: 1rem;
  text-align: left;
  padding-left: 1rem;
  width: 17rem;
  border: 1px solid rgb(255, 165, 0);
}

.card-button {
  display: block;
  float: right;
  width: 3.5rem;
  border-style: none;
  border-left: 1px solid rgb(255, 165, 0);
  border-bottom: 1px solid rgb(255, 165, 0);
}

.active {
  background-color: red;
}

.card-text {
  margin-top: 4rem;
  p {
    margin: 1rem 0;
  }
}

