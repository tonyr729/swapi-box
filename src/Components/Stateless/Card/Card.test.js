import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card', () => {

  describe('Component', () => {
    it('should match snapshot', () => {
      const card = renderer.create(<Card data={{vehicleClass: "awesome"}}/>).toJSON();

      expect(card).toMatchSnapshot();
    })

    it('should not pass the favorites property in to be rendered', () => {
      const wrapper = shallow(<Card data={{favorite: true}} />)

      expect(wrapper).toMatchSnapshot();
    })

    it('should call setFavorites when clicked', () => {
      const setFavorites = jest.fn()
      const wrapper = shallow(<Card data={{name: "Tony"}} setFavorites={ setFavorites }/>)
      
      wrapper.find('button').simulate('click');

      expect(setFavorites.mock.calls.length).toEqual(1)
    })
  })
})