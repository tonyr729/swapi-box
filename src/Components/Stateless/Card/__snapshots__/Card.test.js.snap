// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should match snapshot 1`] = `
<div
  className="card"
>
  <button
    className="card-button active"
    onClick={[Function]}
  >
    ★
  </button>
  <div
    className="card-text"
  >
    <p>
      Class
      : 
      awesome
    </p>
  </div>
</div>
`;

exports[`Card Component should not pass the favorites property in to be rendered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    data={
        Object {
            "favorite": true,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="card-button active"
          onClick={[Function]}
>
          ★
</button>,
        <div
          className="card-text"
>
          
</div>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "★",
          "className": "card-button active",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "★",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
          ],
          "className": "card-text",
        },
        "ref": null,
        "rendered": Array [
          undefined,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="card-button active"
            onClick={[Function]}
>
            ★
</button>,
          <div
            className="card-text"
>
            
</div>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "★",
            "className": "card-button active",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "★",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
            ],
            "className": "card-text",
          },
          "ref": null,
          "rendered": Array [
            undefined,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
